File: Rentacar.API/Rentacar.Entities/Booking.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace Rentacar.Entities
{
    public partial class Booking
    {
        public int BookingId { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public int UserId { get; set; }
        public int VehicleId { get; set; }
        public decimal? TotalPrice { get; set; }
        public bool ReviewAdded { get; set; }

        public virtual User User { get; set; }
        public virtual Vehicle Vehicle { get; set; }
    }
}

File: rentacar.flutter/lib/models/booking.dart
import 'dart:convert';

import 'package:rentacar/models/cardInfo.dart';

class Booking {
  int bookingId = 0;
  DateTime? startDate;
  DateTime? endDate;
  DateTime? createdDate;
  DateTime? updatedDate;
  int userId = 0;
  int vehicleId = 0;
  double? totalPrice = 0;
  CardInfo? cardInfo;

  Booking(
      this.bookingId,
      this.startDate,
      this.endDate,
      this.createdDate,
      this.updatedDate,
      this.userId,
      this.vehicleId,
      this.totalPrice,
      this.cardInfo);

  Booking.fromJson(Map<String?, dynamic> bookingMap) {
    bookingId = bookingMap['bookingId'];
    startDate = DateTime.parse(bookingMap['startDate']);
    endDate = DateTime.parse(bookingMap['endDate']);
    createdDate = DateTime.parse(bookingMap['createdDate']);
    updatedDate = DateTime.parse(bookingMap['updatedDate']);
    userId = bookingMap['userId'];
    vehicleId = bookingMap['vehicleId'];
    totalPrice = bookingMap['totalPrice'];
    cardInfo = CardInfo?.fromJson(bookingMap['cardInfo']);
  }

  Map<String, dynamic> toJson() {
    return {
      'bookingId': bookingId,
      'startDate': startDate?.toIso8601String(),
      'endDate': endDate?.toIso8601String(),
      'createdDate': createdDate?.toIso8601String(),
      'updatedDate': updatedDate?.toIso8601String(),
      'userId': userId,
      'vehicleId': vehicleId,
      'totalPrice': totalPrice,
      'cardInfo': cardInfo?.toJson()
    };
  }
}
